ONSdigital/census-rm-pubsub-adapter
# Processor Restarts

# Motivation and Context
Enable processor restarts on errors from the rabbit channels and pubsub receiver so that the entire app is not immediatelly killed.

# What has changed
* Attempt to restart processors on errors on the error channel
* Send rabbit channel errors to the error channel
* Introduce a short delay on showing ready to give the processors time to actually attempt connection
* Each processor keeps its own context for nicer processor stopping (child of the global app context)

# How to test?
Run the app, try killing a rabbit connection or deleting a pubsub subscription it uses, it should attempt restart.

# Links
https://trello.com/c/ObrXF2L5/1012-tech-session-restart-pubsub-processors
